llvm/lib/CodeGenData/OutlinedHashTreeRecord.cpp

```diff
@@ -119,8 +119,8 @@ void OutlinedHashTreeRecord::convertToStableData(
       [&NodeIdMap](const HashNode *Current) {
         size_t Index = NodeIdMap.size();
         NodeIdMap[Current] = Index;
-        assert(Index = NodeIdMap.size() + 1 &&
-                       "Expected size of NodeMap to increment by 1");
+        assert(Index + 1 == NodeIdMap.size() &&
+               "Expected size of NodeMap to increment by 1");
```

Comment:
Nit: Parens and perhaps a message something like below would give clearer hint on the intent of the assert.
```
        assert((Index + 1 == NodeIdMap.size()) &&
               "Expected 'Current' to not already exist in 'NodeIdMap'");
```